alexandria/core/filters.py

```diff
@@ -132,6 +136,25 @@ def filter(self, qs, value):
         return qs.filter(filters)
 
 
+class FullTextSearchFilter(CharFilter):
+    def filter(self, qs, value):
+        if value in EMPTY_VALUES:
+            return qs
+
+        search_configs = {
+            settings.ISO_639_TO_PSQL_SEARCH_CONFIG.get(lang, "simple")
+            for lang, desc in settings.LANGUAGES
+        }
+
+        search_queries = [
+            SearchQuery(value, config=config, search_type="phrase")
+            for config in search_configs
+        ]
+        search_query = reduce(or_, search_queries)
```

Comment:
I think you can improve the performance quite a bit if you pair each `SearchQuery` with an additional `Q` object that matches the file's detected language. That way, you won't search german-stemmed expressions in english text, which also has the benefit of avoiding possible false positives

Reply:
Done